tsconfig.cypress.component.json

```diff
@@ -1,6 +1,6 @@
 {
   "extends": ["./tsconfig.app.json"],
-  "include": ["src", "cypress", "cypress.config.ts"],
+  "include": ["cypress", "cypress.config.ts"],
```

Comment:
Sadly this will not work.

You have to include the src (or specific components) as you render them in component tests.

With your proposed changes, one cannot import the `App` inside the `App.test.component.tsx`.

<img width="1231" alt="image" src="https://github.com/WojciechMatuszewski/multiple-tsconfig-cypress/assets/26322927/2a39259d-962c-43bd-a2e5-5f69169951f8">

Reply:
Okay, I figured it out. The reason why the `App.tsx` uses the `tsconfig.cypress.component.json` is due to how the `references` in `tsconfig` are ordered.

After switching the order to this

```
{
  "references": [
    { "path": "./tsconfig.node.json" },
    { "path": "./tsconfig.cypress.component.json" },
    { "path": "./tsconfig.cypress.e2e.json" },
    { "path": "./tsconfig.app.json" }
  ],
  "files": [],
  // This is needed for Cypress to parse the config file.
  "ts-node": {
    "transpileOnly": true,
    "compilerOptions": {
      "module": "ESNext"
    }
  }
}
```

The type-checking works as expected.

Reply:
TIL I know the order in references matter :D , so your problem is solved?

Reply:
Yes. I'm going to update the main branch in a sec.